src/coreclr/vm/ClrEtwAll.man

```diff
@@ -1775,6 +1787,32 @@
                         </UserData>
                     </template>
 
+                    <template tid="WaitHandleWaitStart">
+                        <data name="WaitSource" inType="win:UInt32" map="WaitHandleWaitSourceMap" />
+                        <data name="AssociatedObjectID" inType="win:Pointer" />
+                        <data name="ClrInstanceID" inType="win:UInt16" />
+                        <UserData>
+                            <WaitHandle xmlns="myNs">
+                                <WaitSource> %1 </WaitSource>
+                                <AssociatedObjectID> %2 </AssociatedObjectID>
+                                <ClrInstanceID> %3 </ClrInstanceID>
+                            </WaitHandle>
+                        </UserData>
+                    </template>
+
+                    <template tid="WaitHandleWaitStop">
+                        <data name="WaitSource" inType="win:UInt32" map="WaitHandleWaitSourceMap" />
+                        <data name="DurationNs" inType="win:Double" />
```

Comment:
I'm skeptical about whether the `DurationNs` is necessary. It was included with the `ContentionStop` event because lock contention durations can indeed be very short. Waits can also be short (~10 microseconds or less), but probably the more interesting cases will be the longer waits. PerfView already computes a duration for start/stop events based on the event timestamps. I think that would be good enough for starters, maybe we can omit the duration in the event until it's deemed necessary.

Reply:
I think as long as we have QPC timestamp in the event data we should be good without the duration. PerfView lacks ability to analyze durations: there's no histograms or even dumping a stacktrace of a particular event (that would be a super useful feature to diagnose long waits like this btw). When I use fields like this I'm usually running custom code with TraceEvent anyway. And QPC precision is pretty much as good as it gets.